From ca9dfe093dcc77a67987cad2a730db59df0a0c03 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Fri, 21 Jul 2023 18:25:40 -0500
Subject: [PATCH] Hardcode Solus .desktop filename

This fixes window association under x11 and Wayland due to the non-standard name that Solus uses for our .desktop file
---
 toolkit/xre/nsAppRunner.cpp | 2 +-
 widget/gtk/nsAppShell.cpp   | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/toolkit/xre/nsAppRunner.cpp b/toolkit/xre/nsAppRunner.cpp
index 79baf84160..093d0226db 100644
--- a/toolkit/xre/nsAppRunner.cpp
+++ b/toolkit/xre/nsAppRunner.cpp
@@ -4745,7 +4745,7 @@ int XREMain::XRE_mainStartup(bool* aExitFlag) {
   // consistently.
 
   // Set program name to the one defined in application.ini.
-  g_set_prgname(gAppData->remotingName);
+  g_set_prgname("net.thunderbird.Thunderbird");
 
   // Initialize GTK here for splash.
 
diff --git a/widget/gtk/nsAppShell.cpp b/widget/gtk/nsAppShell.cpp
index 0b3c0e18d1..56e001ba31 100644
--- a/widget/gtk/nsAppShell.cpp
+++ b/widget/gtk/nsAppShell.cpp
@@ -295,7 +295,7 @@ nsresult nsAppShell::Init() {
       // Only bother doing this for the parent process, since it's the one
       // creating top-level windows.
       if (gAppData) {
-        gdk_set_program_class(gAppData->remotingName);
+        gdk_set_program_class("net.thunderbird.Thunderbird");
       }
     }
   }
-- 
2.41.0

